### DISCUSSION TITLE
Should quadratic votes have to pass through the round contract to the voting contract?

### DISCUSSION THREAD

**Discussion by nfrgosselin** [gitcoinco/grants-stack#338] [Q&A] [answered]
The current design of the protocol requires that the voting front end sends votes to the round contract, which then passes the information to the voting contract via the `vote` function. The benefit of this is that we get an extra layer of security in that the voting contract will only accept votes coming from the valid round contract. 

Does this make sense / are there reasons we shouldn't do this?

**Comment by nfrgosselin** [comment]
Recording a discussion between @hmrtn and @thelostone-mc in Discord:

@hmrtn:
I think it makes sense - RoundMgmt.vote(..) ==> Strategy.vote(...).  It might be a little redundant if its not entirely necessary, though. Would the round manager store these votes somewhere?

@thelostone-mc:
This depends on if we have 1 QF voting contract per round and if all rounds doing QF use the same QF voting contract 
The pass-through makes it clear that the vote was casted for the round and ensure a gate-keeper of sorts irrespective of the route we go 
Could you clarify what benefits we'd stand to gain if we go by letting folks directly invoke votingStrategy contract?

**Comment by hansmrtn** [comment]
Thanks, @nfrgosselin. 

@thelostone-mc I think isolation and customizability are the benefits of each round having its own `votingStrategy` contract. The isolated instances per round could simplify the necessary data models and indexing of votes, too!

**Reply by thelostone-mc** [reply]
Oh you would still have that here as the round is acting as a gatekeeper and nothing beyond that. 
The voting Contract unique to every round -> would provide the data required to index votes and stuff.

In terms of isolation -> I do feel having a line of connection between the round and voting contract makes sense cause every round would have one voting contract despite the added cost of doing the extra hop for casting a vote. 

If we removed the vote dependency from the round contract, and folks could directly interact wth the voting contract and we'd have to duplicate round start and round end. 

@gravityblast adding you to keep you in loop as well

**Comment by nfrgosselin** [comment, marked as answer]
Just documenting for posterity: we will continue to have the votes pass through the round contract.

@thelostone-mc @hmrtn
